### ISSUE TITLE
[Snowflake]: Changed place of 'NOT' in 'WHERE' statement

### ISSUE THREAD

**Issue by ajuszczak** [tobymao/sqlglot#3890]
SQLGlot implicitly changes the place, where 'NOT' is put in 'WHERE' statement. Such change, may result in different output of the query. I would assume that with the same dialect provided on both ends, the same query (logically) would be provided. 

Code to reproduce:
```
from sqlglot import parse_one

if __name__ == '__main__':
    before_sql = "SELECT * FROM table WHERE column NOT IN (1, 2, 3)"
    expression = parse_one(before_sql, dialect='snowflake')
    after_sql = expression.sql(dialect='snowflake')

    assert before_sql == after_sql, f"\n{before_sql}\nIS NOT THE SAME AS\n{after_sql}"
```
Output:
```
AssertionError: 
SELECT * FROM table WHERE column NOT IN (1, 2, 3)
IS NOT THE SAME AS
SELECT * FROM table WHERE NOT column IN (1, 2, 3)
```

Screenshot:
<img width="1143" alt="Zrzut ekranu 2024-08-9 o 13 35 33" src="https://github.com/user-attachments/assets/d45472b0-68c7-4310-8a22-66eb163f1ae2">

Official Snowflake documentation does not cover that case.

**Comment by georgesittas** [comment]
This looks like a snowflake bug imho, especially if not documented. @tobymao if I recall correctly we had a discussion about this in slack? Do you remember the verdict around this?

**Comment by ajuszczak** [comment]
The only explanation I've found is this: https://community.snowflake.com/s/article/Behaviour-of-NOT-IN-with-NULL-values but I'm not sure it could be treated as 'official documentation'. 

When it comes to behavior of SQLGlot on other dialects I did not test it, neither I ran query provided above on other databases.

When it comes to why it happens in SQLGlot, in my opinion it is somewhere around here: `sqlglot.generator.Generator.not_sql`. On the AST, for the query I mentioned we would see: `Where<Not<In<Column>>>`. When converting `Not` to `String`, it always assumes `Not` should be before nested expression, while `In` always assumes that nested expression is before `In`.

Code Snippet:
`return f"{self.sql(expression, 'this')}{is_global} IN {in_sql}"` - `in_sql` method.
`return f"NOT {self.sql(expression, 'this')}"` - `not_sql` method.

I saw a lot of cases where 'NOT' was used like: '<column> NOT IN', '<column> IS NOT NULL', '<column> IS NOT BETWEEN', so maybe changing the default behavior to always provide it like mentioned before would (maybe not solve the issue, but) cause an issue less apparent?

If such change is not acceptable in library itself, I would appreciate guidance on how that could be achieved on my end only.
